### PR TITLE
refactor(website): consolidate styling for buttons

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

In addition to consolidating button styles this PR also redesigns the install text on the landing page.

### <samp>🤖 Generated by Copilot at 68244da</samp>

*  Add `buttonVariants` function to `Button.tsx` to generate class names for different button styles using `class-variance-authority` library ([link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-66a700d82af7c27efdb24a07c1c53e2c79464c3c1fbae815f879d3f283076730R1-R17))
* Import `buttonVariants` function in `page.tsx` files for home, packages, and package pages and use it to replace long and repetitive class names for link elements with shorter and consistent ones ([link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-64a45b4e89f8fbf788c8687deda76281619fbd0e1a341c61338e79df1aa02179R6), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-e3cacbb50270c1d7ee35c59831191a39a16327b6ce29e9167d17f0274f5fa546R6), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-149efc25407f13009a436bda2625f7223097d1c1f85adc6d78faecccec0105c0R7), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-64a45b4e89f8fbf788c8687deda76281619fbd0e1a341c61338e79df1aa02179L34-R35), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-64a45b4e89f8fbf788c8687deda76281619fbd0e1a341c61338e79df1aa02179L48-R49), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-e3cacbb50270c1d7ee35c59831191a39a16327b6ce29e9167d17f0274f5fa546L31-R32), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-e3cacbb50270c1d7ee35c59831191a39a16327b6ce29e9167d17f0274f5fa546L54-R55), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-e3cacbb50270c1d7ee35c59831191a39a16327b6ce29e9167d17f0274f5fa546L69-R67), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-149efc25407f13009a436bda2625f7223097d1c1f85adc6d78faecccec0105c0L22-R27), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-149efc25407f13009a436bda2625f7223097d1c1f85adc6d78faecccec0105c0L43-R35))
* Import `buttonVariants` function in `InstallButton.tsx` and use it to replace class name for button element that copies installation command to clipboard ([link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-61008a0ff1d350835886a22df5398b0a789311b858abe9aca33d41d53d3face3R7), [link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-61008a0ff1d350835886a22df5398b0a789311b858abe9aca33d41d53d3face3L19-R20))
* Add `gap-10` class to container div for title and subtitle of home page to add some spacing between them ([link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-149efc25407f13009a436bda2625f7223097d1c1f85adc6d78faecccec0105c0L15-R16))
* Remove commented out code for guide link on home page, which is not used ([link](https://github.com/discordjs/discord.js/pull/9466/files?diff=unified&w=0#diff-149efc25407f13009a436bda2625f7223097d1c1f85adc6d78faecccec0105c0L22-R27))

### CODE REVIEW

File: apps/website/src/app/page.tsx [hunk 1]
```diff
@@ -12,35 +13,26 @@ export default function Page() {
 			<Banner />
 			<div className="mx-auto max-w-6xl flex flex-col place-items-center gap-24 px-8 pb-16 pt-12 lg:min-h-[calc(100vh_-_40px)] lg:place-content-center lg:py-10">
 				<div className="flex flex-col place-items-center gap-10 lg:flex-row lg:gap-6">
-					<div className="flex flex-col place-items-center gap-8 text-center">
+					<div className="flex flex-col place-items-center gap-8 text-center space-y-xl">
```
Comment (iCrawl) [inline apps/website/src/app/page.tsx:16]: Don't use the space utility, and increase the gap instead.

File: apps/website/src/styles/button.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { cva } from 'class-variance-authority';
+
+export const button = cva(
```
Comment (iCrawl) [inline apps/website/src/styles/button.tsx:3]: Call this file `Button.tsx` and this function `buttonVariants`

File: apps/website/src/styles/button.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	'h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center rounded px-6 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white gap-2',
+	{
+		variants: {
+			intent: {
```
Comment (iCrawl) [inline apps/website/src/styles/button.tsx:7]:
```suggestion
			variant: {
```

File: apps/website/src/styles/button.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+					'focus:ring',
+					'focus:ring-width-2',
+					'focus:ring-blurple',
+					'dark:active:bg-dark-200',
+					'dark:hover:bg-dark-300',
+				],
+			},
+		},
+		defaultVariants: {
+			intent: 'primary',
```
Comment (iCrawl) [inline apps/website/src/styles/button.tsx:34]:
```suggestion
			variant: 'primary',
```

File: apps/website/src/styles/button.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { cva } from 'class-variance-authority';
+
+export const button = cva(
+	'h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center rounded px-6 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white gap-2',
+	{
+		variants: {
+			intent: {
+				primary: ['bg-blurple', 'text-white', 'border-0'],
+				secondary: [
```
Comment (iCrawl) [inline apps/website/src/styles/button.tsx:9]: Don't use arrays, just a single string with all classes in it.

File: apps/website/src/components/InstallButton.tsx [hunk 6]
```diff
@@ -16,7 +17,7 @@ export function InstallButton() {
 
 	return (
 		<button
-			className="cursor-copy select-none bg-transparent px-4 py-2 text-sm text-dark-50 dark:text-light-900"
+			className={button({ intent: 'secondary', className: 'cursor-copy font-mono' })}
```
Comment (iCrawl) [inline apps/website/src/components/InstallButton.tsx:20]:
```suggestion
			className={button({ variant: 'secondary', className: 'cursor-copy font-mono' })}
```

File: apps/website/src/app/page.tsx [hunk 7]
```diff
@@ -12,35 +13,26 @@ export default function Page() {
...
+								className={button({ intent: 'secondary' })}
 								href="https://discordjs.guide"
 								rel="noopener noreferrer"
 								target="_blank"
 							>
 								Guide <FiExternalLink />
 							</a>
 							<a
-								className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none appearance-none place-items-center gap-2 border border-light-900 rounded bg-white px-4 text-base font-semibold leading-none text-black no-underline outline-none transition duration-200 active:translate-y-px dark:border-dark-100 hover:border-black active:bg-light-300 dark:bg-dark-400 hover:bg-light-200 dark:text-white focus:ring focus:ring-width-2 focus:ring-blurple dark:active:bg-dark-200 dark:hover:bg-dark-300"
+								className={button({ intent: 'secondary' })}
```
Comment (iCrawl) [inline apps/website/src/app/page.tsx:35]:
```suggestion
								className={button({ variant: 'secondary' })}
```

File: apps/website/src/components/InstallButton.tsx [hunk 8]
```diff
@@ -4,6 +4,7 @@ import { FiCheck } from '@react-icons/all-files/fi/FiCheck';
 import { FiCopy } from '@react-icons/all-files/fi/FiCopy';
 import { useEffect, useState } from 'react';
 import { useCopyToClipboard } from 'react-use';
+import { button } from '~/styles/button';
```
Comment (iCrawl) [inline apps/website/src/components/InstallButton.tsx:7]:
```suggestion
import { buttonVariants } from '~/styles/Button';
```

File: apps/website/src/app/page.tsx [hunk 9]
```diff
@@ -12,35 +13,26 @@ export default function Page() {
...
 							</Link>
-							{/* <Link
-								className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center border-0 rounded bg-blurple px-6 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white"
-								href="/guide"
-							>
-								Guide
-							</Link> */}
 							<a
-								className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center gap-2 border border-light-900 rounded bg-white px-4 text-base font-semibold leading-none text-black no-underline outline-none transition duration-200 active:translate-y-px dark:border-dark-100 hover:border-black active:bg-light-300 dark:bg-dark-400 hover:bg-light-200 dark:text-white focus:ring focus:ring-width-2 focus:ring-blurple dark:active:bg-dark-200 dark:hover:bg-dark-300"
+								className={button({ intent: 'secondary' })}
```
Comment (iCrawl) [inline apps/website/src/app/page.tsx:27]:
```suggestion
								className={buttonVariants({ variant: 'secondary' })}
```

File: apps/website/src/app/page.tsx [hunk 10]
```diff
@@ -12,35 +13,26 @@ export default function Page() {
...
 							The <span className="relative rounded bg-blurple px-3 py-1 text-white">most popular</span> way to build
 							Discord bots.
 						</h1>
 						<p className="my-6 leading-normal text-neutral-700 dark:text-neutral-300">{DESCRIPTION}</p>
 						<div className="flex flex-row gap-4">
-							<Link
-								className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center border-0 rounded bg-blurple px-6 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white"
-								href="/docs"
-							>
+							<Link className={button()} href="/docs">
```
Comment (iCrawl) [inline apps/website/src/app/page.tsx:23]:
```suggestion
							<Link className={buttonVariants()} href="/docs">
```

File: apps/website/src/app/page.tsx [hunk 11]
```diff
@@ -4,6 +4,7 @@ import Link from 'next/link';
 import vercelLogo from '~/assets/powered-by-vercel.svg';
 import { Banner } from '~/components/Banner';
 import { InstallButton } from '~/components/InstallButton';
+import { button } from '~/styles/button';
```
Comment (iCrawl) [inline apps/website/src/app/page.tsx:7]:
```suggestion
import { buttonVariants } from '~/styles/Button';
```

File: apps/website/src/app/docs/packages/page.tsx [hunk 12]
```diff
@@ -66,10 +60,7 @@ export default function Page() {
 					</div>
 				</a>
 			</div>
-			<Link
-				className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center place-self-center gap-2 border-0 rounded bg-blurple px-4 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white"
-				href="/"
-			>
+			<Link className={button({ className: 'place-self-center' })} href="/">
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/page.tsx:63]:
```suggestion
			<Link className={buttonVariants({ className: 'place-self-center' })} href="/">
```

File: apps/website/src/app/docs/packages/page.tsx [hunk 13]
```diff
@@ -51,10 +48,7 @@ export default function Page() {
 						</div>
 					</Link>
 				))}
-				<a
-					className="h-11 flex transform-gpu cursor-pointer select-none appearance-none place-content-between border border-neutral-300 rounded bg-white p-4 text-base font-semibold leading-none text-black outline-none active:translate-y-px dark:border-dark-100 active:bg-neutral-200 dark:bg-dark-400 hover:bg-neutral-100 dark:text-white focus:ring focus:ring-width-2 focus:ring-blurple dark:active:bg-dark-200 dark:hover:bg-dark-300"
-					href="https://discord-api-types.dev/"
-				>
+				<a className={button({ intent: 'secondary' })} href="https://discord-api-types.dev/">
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/page.tsx:51]:
```suggestion
				<a className={buttonVariants({ variant: 'secondary' })} href="https://discord-api-types.dev/">
```

File: apps/website/src/app/docs/packages/page.tsx [hunk 14]
```diff
@@ -27,11 +28,7 @@ export default function Page() {
 					</div>
 				</a>
 				{PACKAGES.map((pkg, idx) => (
-					<Link
-						className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-content-between border border-neutral-300 rounded bg-white p-4 text-base font-semibold leading-none text-black outline-none active:translate-y-px dark:border-dark-100 active:bg-neutral-200 dark:bg-dark-400 hover:bg-neutral-100 dark:text-white focus:ring focus:ring-width-2 focus:ring-blurple dark:active:bg-dark-200 dark:hover:bg-dark-300"
-						href={`/docs/packages/${pkg}`}
-						key={`${pkg}-${idx}`}
-					>
+					<Link className={button({ intent: 'secondary' })} href={`/docs/packages/${pkg}`} key={`${pkg}-${idx}`}>
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/page.tsx:31]:
```suggestion
					<Link className={buttonVariants({ variant: 'secondary' })} href={`/docs/packages/${pkg}`} key={`${pkg}-${idx}`}>
```

File: apps/website/src/app/docs/packages/page.tsx [hunk 15]
```diff
@@ -3,6 +3,7 @@ import { VscArrowLeft } from '@react-icons/all-files/vsc/VscArrowLeft';
 import { VscArrowRight } from '@react-icons/all-files/vsc/VscArrowRight';
 import { VscPackage } from '@react-icons/all-files/vsc/VscPackage';
 import Link from 'next/link';
+import { button } from '~/styles/button';
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/page.tsx:6]:
```suggestion
import { buttonVariants } from '~/styles/Button';
```

File: apps/website/src/app/docs/packages/[package]/page.tsx [hunk 16]
```diff
@@ -45,10 +46,7 @@ export default async function Page({ params }: { params: { package: string } })
 					</Link>
 				)) ?? null}
 			</div>
-			<Link
-				className="h-11 flex flex-row transform-gpu cursor-pointer select-none appearance-none place-items-center place-self-center gap-2 border-0 rounded bg-blurple px-4 text-base font-semibold leading-none text-white no-underline outline-none active:translate-y-px focus:ring focus:ring-width-2 focus:ring-white"
-				href="/docs/packages"
-			>
+			<Link className={button({ className: 'place-self-center' })} href="/docs/packages">
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/[package]/page.tsx:49]:
```suggestion
			<Link className={buttonVariants({ className: 'place-self-center' })} href="/docs/packages">
```

File: apps/website/src/app/docs/packages/[package]/page.tsx [hunk 17]
```diff
@@ -31,7 +32,7 @@ export default async function Page({ params }: { params: { package: string } })
 			<div className="flex flex-col gap-4">
 				{data.map((version, idx) => (
 					<Link
-						className="h-11 flex flex-col transform-gpu cursor-pointer select-none appearance-none place-content-center border border-neutral-300 rounded bg-white p-4 text-base font-semibold leading-none text-black outline-none active:translate-y-px dark:border-dark-100 active:bg-neutral-200 dark:bg-dark-400 hover:bg-neutral-100 dark:text-white focus:ring focus:ring-width-2 focus:ring-blurple dark:active:bg-dark-200 dark:hover:bg-dark-300"
+						className={button({ intent: 'secondary' })}
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/[package]/page.tsx:35]:
```suggestion
						className={buttonVariants({ variant: 'secondary' })}
```

File: apps/website/src/app/docs/packages/[package]/page.tsx [hunk 18]
```diff
@@ -3,6 +3,7 @@ import { VscArrowRight } from '@react-icons/all-files/vsc/VscArrowRight';
 import { VscVersions } from '@react-icons/all-files/vsc/VscVersions';
 import Link from 'next/link';
 import { notFound } from 'next/navigation';
+import { button } from '~/styles/button';
```
Comment (iCrawl) [inline apps/website/src/app/docs/packages/[package]/page.tsx:6]:
```suggestion
import { buttonVariants } from '~/styles/Button';
```

**Review by Jiralite — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**